packages/core/lib/segments/segment.js

```diff
@@ -280,16 +280,19 @@ Segment.prototype.removeSubsegment = function removeSubsegment(subsegment) {
  * Adds error data into the segment.
  * @param {Error|string} err - The error to capture.
  * @param {boolean} [remote] - Flag for whether the exception caught was remote or not.
+ * @param {boolean} [fault] - Flag for whether the fault flag should be set.
  */
 
-Segment.prototype.addError = function addError(err, remote) {
+Segment.prototype.addError = function addError(err, remote, fault) {
```

Comment:
Changing this function signature will be a breaking / backwards-incompatible change. Ideally, we would want to maintain the current expected behaviour so that this change does not cause any unexpected changes for others as well. 

I would recommend setting a default value of `true` for the `fault` parameter to ensure that the fault flag is always added by default, and would only be excluded if a third argument with the value of `false` is included. 

I am also curious, what is the use case for excluding the fault flag? Is there a specific reason why it would not be included?

Reply:
@carolabadeer  Here I would like to answer, as I am having the same need as @icholy. In the segments error [documentation](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-errors), is mentioned that the flags `error`, `throttle` and `fault`, can represent different types of errors. 
In my personal use case, we are flagging all expected errors as `error` and unexpected errors as `fault`.
At the moment of catching and handling the error logic, we want to add error data to the segment with `addError` method, but it always set `fault` flag to `true`. So we need to overwrite the flag accordingly, whether the error is expected or unexpected.

An extra param as @icholy suggest will reduce extra code, and avoid overwriting the segment properties directly.

Reply:
@carolabadeer the change _is_ backwards compatible. The `fault || fault === undefined` condition is semantically equivalent to defaulting `fault` to true.